### PR TITLE
We create /System/Library/LaunchDaemons/ for builds that don't install plists in that path

### PR DESCRIPTION
**Pull request by msaboff**
#### d0b2da3599507511dd695907e342cc04da50d9cf
<pre>
We create /System/Library/LaunchDaemons/ for builds that don&apos;t install plists in that path
<a href="https://bugs.webkit.org/show_bug.cgi?id=242081">https://bugs.webkit.org/show_bug.cgi?id=242081</a>
&lt;rdar://96013133&gt;

Reviewed by Saam Barati.

The Output Files for the Copy launchd plist and create symlink script phases for the adattributed and webpushd
targets will create the containing folder of any of the files listed there, even though the script creates the
containing directory and file when needed.  Therefore we changed the Output Files specified to have a computed
path value when installing a plist, or an empty value when we aren&apos;t.  This output path value is based on the
SKIP_INSTALL configuration variable.

* Source/WebKit/Configurations/adattributiond.xcconfig:
* Source/WebKit/Configurations/webpushd.xcconfig:
* Source/WebKit/Scripts/copy-launchd-plist-and-create-symlink.sh:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/251969@main">https://commits.webkit.org/251969@main</a>
</pre>

### CODE REVIEW

**Comment by davidquesada** [comment]
Looks good to me. I think it would also be possible to resolve this by giving the script phases an output path that's defined by one build setting that evaluates to an empty string when the plists won't be installed. But as you mentioned, that isn't really important since the plists are "leaf nodes", and no other operation will ever need to block on creating them.

**Review by saambarati — APPROVED** [review]
r=me

**Review by emw-apple — COMMENTED** [review]
If we remove all output paths, the script always runs. I know this script is relatively fast, but as I'm trying to improve incremental build performance, this would be a (small) step in the wrong direction.

I would eventually like to see these scripts rewritten as a build rule, because then—among other benefits—we could set `EXCLUDED_SOURCE_FILE_NAMES = com.apple.webpushd.plist` and prevent the script from ever executing. But in the meantime, could we do something like what @davidquesada is suggesting?

**Comment by msaboff** [comment]
Testing a version where the output path is defined by one build setting that evaluates to an empty string when the plists won't be installed as @davidquesada  suggested.

**Comment by msaboff** [comment]
@emw-apple and @davidquesada Updated the patch as suggested and tested it locally.

File: Source/WebKit/WebKit.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -16153,7 +16153,7 @@
 			outputFileListPaths = (
 			);
 			outputPaths = (
-				"$(DSTROOT)$(LAUNCHD_PLIST_INSTALL_PATH)/$(LAUNCHD_PLIST_FILE_NAME)",
+				"$(DERIVED_FILE_DIR)/newOutputFile",
```
Comment (emw-apple) [inline Source/WebKit/WebKit.xcodeproj/project.pbxproj:16156]: I think this is supposed to be:
```suggestion
				"$(LAUNCHD_PLIST_OUTPUT_FILE)",
```

Comment (msaboff) [reply Source/WebKit/WebKit.xcodeproj/project.pbxproj:16156]: Agreed.  I made that change, so I don't know why it didn't take.  I'll post an updated PR.

Comment (emw-apple) [reply Source/WebKit/WebKit.xcodeproj/project.pbxproj:16156]: r=me once this is updated!

Comment (msaboff) [reply Source/WebKit/WebKit.xcodeproj/project.pbxproj:16156]: Done.

**Pull request merged**